Util/channel/src/main/java/io/deephaven/util/channel/SeekableChannelsProvider.java

```diff
@@ -18,26 +18,69 @@
 public interface SeekableChannelsProvider extends SafeCloseable {
 
     /**
-     * Take the file source path or URI and convert it to a URI object.
+     * Take the file source path or URI string and convert it to a URI object.
      *
      * @param source The file source path or URI
      * @return The URI object
      */
     static URI convertToURI(final String source) {
+        if (source.isEmpty()) {
+            throw new IllegalArgumentException("Cannot convert empty source to URI");
+        }
         final URI uri;
         try {
             uri = new URI(source);
         } catch (final URISyntaxException e) {
             // If the URI is invalid, assume it's a file path
-            return new File(source).toURI();
+            return convertFileToURI(new File(source));
         }
         if (uri.getScheme() == null) {
-            // Need to convert to a "file" URI
-            return new File(source).toURI();
+            // Convert to a "file" URI
+            return convertFileToURI(new File(source));
         }
         return uri;
     }
 
+    /**
+     * Takes a file and convert it to a URI object. We should use this method instead of {@link File#toURI()} because
+     * {@link File#toURI()} internally calls {@link File#isDirectory()}, which can lead to disk access.
+     *
+     * @param file The file
+     * @return The URI object
+     */
+    static URI convertFileToURI(final File file) {
+        String absPath = file.getAbsolutePath();
+        if (absPath.isEmpty()) {
+            throw new IllegalArgumentException("Cannot convert file with empty path to URI: " + file);
+        }
```

Comment:
I kept it as a sanity check and second because it helps with my check later around absPath.charAt(0), just to make sure there is a 0. 
If you think I can remove it, I will.

Reply:
I expect it's impossible for an absolute path, but the check isn't harmful.